File: backend/versuion/src/main/java/com/example/versuion/models/Fournisseur.java
package com.example.versuion.models;

import lombok.*;

import javax.persistence.Column;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.OneToMany;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Entity
public class Fournisseur extends AbstractEntity{

    private String nom;

    private String prenom;

    @Embedded
    private Adresse adresse;

    private String photo;

    private String mail;

    private String numTel;

    @Column(name = "identreprise")
    private Integer idEntreprise;

    @OneToMany(mappedBy = "Fournisseur")
    private List<ComandeFournisseur> comandeFournisseurList;
}

File: frontend/my-first-project/src/app/Models/Fournisseur.ts
import { AdresseDto } from "./AdresseDto";
import { CommandeFournisseur } from "./CommandeFournisseur";

export interface Fournisseur {
  id?: number;
  creationDate?: number;
  lastModifiedDate?: number;
  nom?: string;
  prenom?: string;
  adresse?: AdresseDto;
  photo?: string;
  mail?: string;
  numTel?: string;
  idEntreprise?: number;
  commandeFournisseurs?: Array<CommandeFournisseur>;
}
